src/channels.rs

```diff
@@ -150,6 +167,65 @@ impl MiEI {
         }
     }
 
+    pub async fn move_course<'a>(
+        &mut self,
+        course: &'a str,
+        new_year: &str,
+        new_semester: &str,
+        new_name: Option<&'a str>,
+        ctx: &Context,
+        guild: GuildId,
+    ) -> serenity::Result<&'a str> {
+        let upper_new_name = new_name.map(|n| n.to_uppercase());
+        if upper_new_name
+            .as_ref()
+            .filter(|r| self.role_exists(r))
+            .is_some()
+        {
+            Err(serenity::Error::Other("Role already exists"))
+        } else if let Some(x) = self.courses.values_mut().find_map(|x| x.pop_role(course)) {
```

Comment:
rename `x` to `old_course` or `course`? It's the only single name variable I found hard to read :sweat_smile: